packages/@react-aria/utils/src/useSyntheticBlurEvent.ts

```diff
@@ -0,0 +1,204 @@
+      stateRef.current.onBlur?.(new SyntheticBlurEvent('blur', e));
+    }
+
+    // We no longer need the MutationObserver once the target is blurred.
+    if (stateRef.current.observer) {
+      stateRef.current.observer.disconnect();
+      stateRef.current.observer = null;
+    }
+  }, [stateRef]);
+
+  // This function is called during a React onFocus event.
+  return useCallback((e: ReactFocusEvent) => {
+
+    let target = e.target as HTMLElement;
+    stateRef.current.isFocused = true;
+
+    if (stateRef.current.target && target !== stateRef.current.target) {
+      stateRef.current.target.removeEventListener('focusout', onBlurHandler);
+      domNodeRemovedObserver.removeTarget(stateRef.current.target);
+      target.addEventListener('focusout', onBlurHandler, {once: true});
+    } else if (!stateRef.current.target || e.target !== stateRef.current.target) {
+      target.addEventListener('focusout', onBlurHandler, {once: true});
+    }
+
+    stateRef.current.target = target;
+    // React does not fire onBlur when an element is disabled. https://github.com/facebook/react/issues/9142
+    // Most browsers fire a native focusout event in this case, except for Firefox. In that case, we use a
+    // MutationObserver to watch for the disabled attribute, and dispatch these events ourselves.
+    // For browsers that do, focusout fires before the MutationObserver, so onBlur should not fire twice.
+    if (
+      e.target instanceof HTMLButtonElement ||
+      e.target instanceof HTMLInputElement ||
+      e.target instanceof HTMLTextAreaElement ||
+      e.target instanceof HTMLSelectElement
+    ) {
+
+      stateRef.current.observer = new MutationObserver(() => {
+        if (stateRef.current.isFocused && (target as any).disabled) {
+          stateRef.current.observer.disconnect();
+          target.dispatchEvent(new FocusEvent('blur'));
+          target.dispatchEvent(new FocusEvent('focusout', {bubbles: true}));
+        }
+      });
+
+      stateRef.current.observer.observe(target, {attributes: true, attributeFilter: ['disabled']});
+    }
+
+    // Some browsers do not fire onBlur when the document.activeElement is removed from the dom.
+    // Firefox has had a bug open about it for 13 years https://bugzilla.mozilla.org/show_bug.cgi?id=559561
+    // A Safari bug has been logged for it as well https://bugs.webkit.org/show_bug.cgi?id=243749
+    domNodeRemovedObserver.setTarget(target);
+  }, []);
+}
+
+
+class DOMNodeRemovedObserver {
+  private observer: MutationObserver | null;
+  private target: HTMLElement | null;
+
+  constructor() {
+    // don't make an observer in SSR (MutationObserver is not supported in Node)
+    if (typeof window !== 'undefined') {
+      this.observer = new MutationObserver((mutationList) => {
+        if (this.target) {
+          for (const mutation of mutationList) {
+            for (const node of mutation.removedNodes) {
+              if (node?.contains?.(this.target)) {
```

Comment:
Small perf optimization could be to do `if (!document.body.contains(this.target)` rather than looping through all the mutations. There could potentially be a lot of mutations at once.